import `boto3` and create a configuration session to specify aws access info


In [1]:
import boto3

session = boto3.Session(
    profile_name="boto3_101"
)  # accessing aws via boto3 using a specific profile (aws user) in `~/.aws/credentials` file

In [2]:
s3 = session.resource("s3")  # grabbing the s3 resource

In [3]:
list(s3.buckets.all())  # iterating through s3 buckets

[s3.Bucket(name='add-demo-cloudfront-v1'),
 s3.Bucket(name='add-demo-s3-v1'),
 s3.Bucket(name='sagemaker-studio-891377239273-vees33vfkeb'),
 s3.Bucket(name='sagemaker-us-east-1-891377239273'),
 s3.Bucket(name='terraform-backend-frz9g')]

In [5]:
for bucket in s3.buckets.all():
    print(bucket.name)  # using for loop to iterate and print bucket name attribute

add-demo-cloudfront-v1
add-demo-s3-v1
sagemaker-studio-891377239273-vees33vfkeb
sagemaker-us-east-1-891377239273
terraform-backend-frz9g


#### List all available `boto3` services


In [6]:
services = session.get_available_services()
print(len(services))
services

404


['accessanalyzer',
 'account',
 'acm',
 'acm-pca',
 'amp',
 'amplify',
 'amplifybackend',
 'amplifyuibuilder',
 'apigateway',
 'apigatewaymanagementapi',
 'apigatewayv2',
 'appconfig',
 'appconfigdata',
 'appfabric',
 'appflow',
 'appintegrations',
 'application-autoscaling',
 'application-insights',
 'application-signals',
 'applicationcostprofiler',
 'appmesh',
 'apprunner',
 'appstream',
 'appsync',
 'apptest',
 'arc-zonal-shift',
 'artifact',
 'athena',
 'auditmanager',
 'autoscaling',
 'autoscaling-plans',
 'b2bi',
 'backup',
 'backup-gateway',
 'backupsearch',
 'batch',
 'bcm-data-exports',
 'bcm-pricing-calculator',
 'bedrock',
 'bedrock-agent',
 'bedrock-agent-runtime',
 'bedrock-data-automation',
 'bedrock-data-automation-runtime',
 'bedrock-runtime',
 'billing',
 'billingconductor',
 'braket',
 'budgets',
 'ce',
 'chatbot',
 'chime',
 'chime-sdk-identity',
 'chime-sdk-media-pipelines',
 'chime-sdk-meetings',
 'chime-sdk-messaging',
 'chime-sdk-voice',
 'cleanrooms',
 'cleanro

`404` `boto3` services available in total. That's nuts!


Checking to see if I have any active `SQS` queues in my AWS account


In [7]:
sqs = session.resource("sqs")
list(sqs.queues.all())

[]